app/controllers/meetings_controller.rb

```diff
@@ -45,58 +26,40 @@ def comment
 
   # rubocop:disable MethodLength
   def delete_comment
-    comment_exists = Comment.where(id: params[:commentid]).exists?
-    is_my_comment = Comment.where(
-      id: params[:commentid],
-      comment_by: current_user.id
-    ).exists?
+    comment_exists = Comment.exists?(id: params[:commentid])
+    is_my_comment = Comment.user_comment?(params[:commentid], current_user)
 
     if comment_exists
-      meeting_id = Comment.where(id: params[:commentid]).first.commentable_id
-      is_my_meeting = MeetingMember.where(
-        meeting_id: meeting_id,
-        user_id: current_user.id,
-        leader: true
-      ).exists?
-      is_member = MeetingMember.where(
-        meeting_id: meeting_id,
-        user_id: current_user.id
-      ).exists?
-    else
-      is_my_meeting = false
-      is_member = false
+      meeting_id = Comment.find(params[:commentid]).commentable_id
+      meeting = Meeting.find(meeting_id)
+      is_my_meeting = MeetingMember.leader?(current_user, meeting)
+      is_member = MeetingMember.member?(current_user, meeting)
     end
 
     if comment_exists && ((is_my_comment && is_member) || is_my_meeting)
-      CommentNotificationsService.remove(comment_id: params[:commentid],
-                                         model_name: 'meeting')
+      remove_notification
```

Comment:
correct me if I'm wrong but in this action, we want to delete a `comment` if it exists AND the comment is the current_user's comment AND the current_user is a member of the meeting right?

Reply:
That sounds correct to me ^

Reply:
I am not sure what to do with this comment. Is there something I should add or change?

Reply:
Sorry for the confusion, I'm just trying to clarify what this action is supposed to do as I feel there's definitely room for improvements. I apologise that I'm unable to point it out right.

Will let you know once I got an idea and if this PR isn't merged by then.